packages/Php/src/Rector/Closure/ClosureToArrowFunctionRector.php

```diff
@@ -0,0 +1,81 @@
+<?php declare(strict_types=1);
+
+namespace Rector\Php\Rector\Closure;
+
+use PhpParser\Node\Expr\ArrowFunction;
+use PhpParser\Node\Stmt\Return_;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Closure;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * @see https://wiki.php.net/rfc/arrow_functions_v2
+ */
+final class ClosureToArrowFunctionRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Change closure to arrow function', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run($meetups)
+    {
+        return array_filter($meetups, function (Meetup $meetup) {
+            return is_object($meetup);
+        });
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run($meetups)
+    {
+        return array_filter($meetups, fn(Meetup $meetup) => is_object($meetup));
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Closure::class];
+    }
+
+    /**
+     * @param Closure $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isAtLeastPhpVersion('7.4')) {
+            return null;
+        }
+
+        if (count((array) $node->stmts) !== 1) {
+            return null;
+        }
+
+        $arrowFunction = new ArrowFunction();
+        $arrowFunction->params = $node->params;
+        $arrowFunction->returnType = $node->returnType;
+        $arrowFunction->byRef = $node->byRef;
+
+        $expr = $node->stmts[0];
+        if ($expr instanceof Return_) {
+            $expr = $expr->expr;
+        }
```

Comment:
I don't think you should do this refactoring if there is no return. It may convert a void closure in a non-void one.

Reply:
Good point, thanks :+1: